### ISSUE TITLE
Make adding observers to post-game study eaiser etc

### ISSUE THREAD

**Issue by killerducky** [WandererXII/lishogi#657]
Great job on #363! A few suggestions:

- After players click "Study with opponent", make it easy for observers to join. e.g. link in the spectator chat, or glowing button like the opponent gets.
- When observers join, I didn't see any way to give them permissions to chat and/or edit the board position. Looks like this is only available using the "+" for "Study with" method.
- Using the "+" / "Study with" method, I didn't get autocomplete for usernames. I think there is autocomplete for normal study creation? And also there was some suggested names I think that you could just click on.

**Comment by killerducky** [comment]
Ok now for spectating I see the glowing button to join, so that seems good. Also now autocomplete is working, don't know what happened the first time.

So it's just the things about how to give ownership to the players etc left.

**Comment by WandererXII** [comment]
Oh now I also noticed the typeahead not working at first, only after reload. Probably not being initiated right. Will look into it.

**Comment by Anfid** [comment]
I have another suggestion related to this issue. I think it would be nice to have an option to just create a public study where everyone who spectates the game or has a link to the study can participate. I feel like in most cases most people don't expect bad actors to hop in on their studies and cause problems. I'm not quite sure how difficult could it be to implement, but it sounds like a very easy to understand and useful study option under the "+" button.

**Comment by WandererXII** [comment]
Fixing the autocomplete - https://github.com/WandererXII/lishogi/commit/9848003d432402fc9fa2bda08cabe30f86e156ae
Inviting others - https://github.com/WandererXII/lishogi/commit/e7ae4112a201e5b9f311b1bdb3e0fd8780f905c0, https://github.com/WandererXII/lishogi/commit/a76f29b09ae5599abfb54bf764ca11cd3bfb1f15
I believe the button glowing works.

> I have another suggestion related to this issue. I think it would be nice to have an option to just create a public study where everyone who spectates the game or has a link to the study can participate. I feel like in most cases most people don't expect bad actors to hop in on their studies and cause problems. I'm not quite sure how difficult could it be to implement, but it sounds like a very easy to understand and useful study option under the "+" button.

You're probably right about the bad actors, worth considering, I will create a separate issue.
